apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-metricsdk-plugin/src/test/java/co/elastic/apm/agent/otelmetricsdk/AbstractOtelMetricsTest.java

```diff
@@ -0,0 +1,624 @@
+        reporter.reset();
+        invokeSdkForceFlush();
+        verifyAdditionalExportersCalled();
+    }
+
+    @Test
+    public void testAttributeTypes() {
+        Meter testMeter = createMeter("test");
+        LongCounter counter = testMeter.counterBuilder("my_counter").build();
+
+        Attributes attribs = Attributes.builder()
+            .put(stringKey("string_attrib"), "foo")
+            .put(doubleKey("double_attrib"), Double.MAX_VALUE)
+            .put(longKey("long_attrib"), Long.MAX_VALUE)
+            .put(booleanKey("bool_attrib"), false)
+            .put(stringArrayKey("string_array_attrib"), List.of("foo", "bar"))
+            .put(doubleArrayKey("double_array_attrib"), List.of(1.0, Double.MAX_VALUE))
+            .put(longArrayKey("long_array_attrib"), List.of(1L, Long.MAX_VALUE))
+            .put(booleanArrayKey("bool_array_attrib"), List.of(true, false))
+            .build();
+
+        counter.add(42, attribs);
+
+        resetReporterAndFlushMetrics();
+        assertThatMetricSets(reporter.getBytes())
+            .hasMetricsetCount(1)
+            .hasMetricsetWithLabelsSatisfying(Map.of(
+                    "otel_instrumentation_scope_name", "test",
+                    "string_attrib", "foo",
+                    "double_attrib", Double.MAX_VALUE,
+                    "long_attrib", Long.MAX_VALUE,
+                    "bool_attrib", false
+                ), metrics -> metrics
+                    .hasMetricsCount(1)
+                    .containsValueMetric("my_counter", 42)
+            );
+    }
+
+    @Test
+    public void testAttributeNameSanitization() {
+        Meter testMeter = createMeter("test");
+        LongCounter counter = testMeter.counterBuilder("my_counter").build();
+
+        Attributes attributes = Attributes.of(stringKey("please*sanitize.me\""), "dont*sanitize.me\"");
+        counter.add(42, attributes);
+
+        resetReporterAndFlushMetrics();
+        assertThatMetricSets(reporter.getBytes())
+            .hasMetricsetCount(1)
+            .hasMetricsetWithLabelsSatisfying(
+                "otel_instrumentation_scope_name", "test",
+                "please_sanitize_me_", "dont*sanitize.me\""
+                , metrics -> metrics
+                    .hasMetricsCount(1)
+                    .containsValueMetric("my_counter", 42)
+            );
+    }
+
+    @Test
+    public void testTimestampPresent() {
+        Meter testMeter = createMeter("test");
+        LongCounter counter = testMeter.counterBuilder("my_counter").build();
+
+        long timestampMicros = System.currentTimeMillis() * 1000L;
+        counter.add(42);
+
+        resetReporterAndFlushMetrics();
+        assertThatMetricSets(reporter.getBytes())
+            .hasMetricsetCount(1)
+            .first()
+            //check for a range of +- 1 second to be safe against CI slowness / clock inaccuracy
+            .hasTimestampInRange(timestampMicros - 1_000_000L, timestampMicros + 1_000_000L);
```

Comment:
[minor] why not use `Instant.now()` before and after the metric collection and use that as narrower interval boundaries ?